// resort-booking-backend/src/routes/rooms.ts
import express from 'express'

const router = express.Router()

// @desc    Get all rooms
// @route   GET /api/rooms
// @access  Public
router.get('/', async (req, res) => {
  try {
    // TODO: Implement get all rooms logic
    const mockRooms = [
      {
        id: 1,
        name: 'Ocean View Suite',
        type: 'Suite',
        price: 299,
        capacity: 4,
        amenities: ['Ocean View', 'Balcony', 'Mini Bar', 'WiFi'],
        images: ['/images/room1.jpg'],
        available: true
      },
      {
        id: 2,
        name: 'Garden Villa',
        type: 'Villa',
        price: 459,
        capacity: 6,
        amenities: ['Private Garden', 'Pool', 'Kitchen', 'WiFi'],
        images: ['/images/room2.jpg'],
        available: true
      }
    ]

    res.status(200).json({
      success: true,
      count: mockRooms.length,
      data: mockRooms
    })
  } catch (error) {
    res.status(500).json({
      success: false,
      error: 'Server Error'
    })
  }
})

// @desc    Get single room
// @route   GET /api/rooms/:id
// @access  Public
router.get('/:id', async (req, res) => {
  try {
    const { id } = req.params

    // TODO: Implement get single room logic
    res.status(200).json({
      success: true,
      message: `Get room ${id} endpoint - Coming Soon`
    })
  } catch (error) {
    res.status(500).json({
      success: false,
      error: 'Server Error'
    })
  }
})

// @desc    Create room
// @route   POST /api/rooms
// @access  Private/Admin
router.post('/', async (req, res) => {
  try {
    // TODO: Implement create room logic
    res.status(201).json({
      success: true,
      message: 'Create room endpoint - Coming Soon'
    })
  } catch (error) {
    res.status(500).json({
      success: false,
      error: 'Server Error'
    })
  }
})

export { router as roomRoutes }
